internal/services/labservice/lab_service_lab_plan_resource.go

```diff
@@ -0,0 +1,716 @@
+	ShutdownEnabledWhenNotConnected bool                       `tfschema:"shutdown_enabled_when_not_connected"`
+}
+
+type DefaultConnectionProfile struct {
+	ClientRdpAccess labplan.ConnectionType `tfschema:"client_rdp_access"`
+	ClientSshAccess labplan.ConnectionType `tfschema:"client_ssh_access"`
+	WebRdpAccess    labplan.ConnectionType `tfschema:"web_rdp_access"`
+	WebSshAccess    labplan.ConnectionType `tfschema:"web_ssh_access"`
+}
+
+type DefaultNetworkProfile struct {
+	SubnetId string `tfschema:"subnet_id"`
+}
+
+type SupportInfo struct {
+	Email        string `tfschema:"email"`
+	Instructions string `tfschema:"instructions"`
+	Phone        string `tfschema:"phone"`
+	Url          string `tfschema:"url"`
+}
+
+type LabServiceLabPlanResource struct{}
+
+var _ sdk.ResourceWithUpdate = LabServiceLabPlanResource{}
+
+func (r LabServiceLabPlanResource) ResourceType() string {
+	return "azurerm_lab_service_lab_plan"
+}
+
+func (r LabServiceLabPlanResource) ModelObject() interface{} {
+	return &LabServiceLabPlanModel{}
+}
+
+func (r LabServiceLabPlanResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return labplan.ValidateLabPlanID
+}
+
+func (r LabServiceLabPlanResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.LabPlanName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"allowed_regions": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 1,
+			MaxItems: 28,
+			Elem: &pluginsdk.Schema{
+				Type:             pluginsdk.TypeString,
+				ValidateFunc:     location.EnhancedValidate,
+				StateFunc:        location.StateFunc,
+				DiffSuppressFunc: location.DiffSuppressFunc,
+			},
+		},
+
+		"default_auto_shutdown_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"disconnect_delay": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: azValidate.ISO8601Duration,
+					},
+
+					"idle_delay": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: azValidate.ISO8601Duration,
+					},
+
+					"no_connect_delay": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: azValidate.ISO8601Duration,
+					},
+
+					"shutdown_on_disconnect_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+
+					"shutdown_on_idle": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(labplan.ShutdownOnIdleModeUserAbsence),
+							string(labplan.ShutdownOnIdleModeLowUsage),
+							string(labplan.ShutdownOnIdleModeNone),
```

Comment:
You mean add "Default:  string(labplan.ShutdownOnIdleModeNone),"?

Reply:
There are plenty examples of both, however we're moving away from the pattern of exposing `None` as Terraform has it's own null type instead (by omitting the field) - we are also planning to remove `None` in other resources in 4.0.

Also just to clarify, we _remove_ `Default: string(labplan.ShutdownOnIdleModeNone)` since the default of null will mean the same thing. 

In the expand/flatten we will require logic to switch out a null for `None`, but this will result in a better and more consistent user experience at the end.